### DISCUSSION TITLE
Error handling for custom sign in page

### DISCUSSION THREAD

**Discussion by djvid** [nextauthjs/next-auth#1443] [Help] [answered]
Hi all, I have a custom sign in page and I'm facing an issue with error handling for email sign in. There seems to be a bug in the URL when i submit an invalid email. Because of this I'm unable to handle error properly.

Steps:
1. When calling `signIn()` i get redirected to `http://localhost:3000/auth/signin?callbackUrl=http://localhost:3000/`
2. Then I submit an invalid email address,`signIn('email', { email: 'xyz' })` 
3. URL is now `http://localhost:3000/auth/signin?callbackUrl=http://localhost:3000/auth/signin?callbackUrl=http://localhost:3000/&error=EmailSignin` 
4. Again, I submit an invalid email and now the url is `http://localhost:3000/auth/signin?callbackUrl=http://localhost:3000/auth/signin?callbackUrl=http://localhost:3000/auth/signin?callbackUrl=http://localhost:3000/&error=EmailSignin&error=EmailSignin`

Anyone else facing this issue?

**Comment by djvid** [comment, marked as answer]
Sorry guys, after going through the docs again managed to find a solution. Just needed to create a custom error page.

**Reply by balazsorban44** [reply]
You might also be interested in the `redirect: false` option of the `signIn` function! https://next-auth.js.org/getting-started/client#using-the-redirect-false-option!

**Reply by djvid** [reply]
Ahh yes, thank you! Better UX with this method 👍 👍
